tools/toolchains/iar.py

```diff
@@ -43,23 +44,34 @@ def check_executable():
 
     def __init__(self, target, notify=None, macros=None, build_profile=None,
                  build_dir=None):
-        mbedToolchain.__init__(self, target, notify, macros, build_dir=build_dir,
-                               build_profile=build_profile)
-        if target.core == "Cortex-M7F" or target.core == "Cortex-M7FD":
-            cpuchoice = "Cortex-M7"
-        elif target.core.startswith("Cortex-M23"):
-            cpuchoice = "8-M.baseline"
-        elif target.core.startswith("Cortex-M33"):
-            cpuchoice = "8-M.mainline"
-        else:
-            cpuchoice = target.core
+        mbedToolchain.__init__(self, target, notify, macros, build_dir=build_dir,build_profile=build_profile)
```

Comment:
style nit: we usually have spaces after commas
```suggestion
        mbedToolchain.__init__(self, target, notify, macros, build_dir=build_dir, build_profile=build_profile)
```